consensus/core/src/utxo/utxo_collection.rs

```diff
@@ -24,6 +25,12 @@ pub trait UtxoCollectionExtensions {
         F: Fn(&TransactionOutpoint, &UtxoEntry, &UtxoEntry) -> bool;
 }
 
+impl UtxoView for UtxoCollection {
+    fn get(&self, outpoint: &TransactionOutpoint) -> Option<UtxoEntry> {
+        self.get(outpoint).cloned()
```

Comment:
The main actual imp of UtxoView is the store, which cannot return things by ref (since it may not exist in the cache). And regardless, usage of this code always would clone the entry anyway to fill the populated tx etc.